### ISSUE TITLE
[CLI]: WARNING: No AMD GPUs specified

### ISSUE THREAD

**Issue by mdivyanshu97** [wandb/wandb#6757]
![Screenshot 2023-12-21 at 12 10 03](https://github.com/wandb/wandb/assets/149976183/a2154368-24b5-4a55-bb1e-384edc51ba50)
### Describe the bug

I have this weird WARNING: No AMD GPUs specified whenever I log using wandb on my cluster. The cluster has NVIDIA Tesla V100 32GB GPU. This warning keeps printing for every batch  and hence slows the code. Do you know how can I fix this?

```python

```

```shell
![image](https://github.com/wandb/wandb/assets/149976183/45d7a14b-1b86-4c80-832b-a2f75beff85b)

```

### Environment

WandB version: wandb==0.16.1

OS:Ubuntu 20

Python version: 3.8.18

Versions of relevant libraries:
absl-py==2.0.0
antlr4-python3-runtime==4.9.3
appdirs==1.4.4
av==11.0.0
cachetools==5.3.2
certifi==2023.11.17
charset-normalizer==3.3.2
click==8.1.7
contourpy==1.1.1
cycler==0.12.1
docker-pycreds==0.4.0
ffmpeg-python==0.2.0
filelock==3.13.1
fonttools==4.46.0
fsspec==2023.12.1
future==0.18.3
fvcore==0.1.5.post20221221
gitdb==4.0.11
gitpython==3.1.40
google-auth-oauthlib==1.0.0
google-auth==2.23.4
grpcio==1.59.3
h5py==3.10.0
huggingface-hub==0.19.4
idna==3.4
imageio==2.33.0
importlib-metadata==6.8.0
importlib-resources==6.1.1
info-nce-pytorch==0.1.4
iopath==0.1.10
jinja2==3.1.2
joblib==1.3.2
kiwisolver==1.4.5
kornia==0.7.0
lightning-utilities==0.10.0
lpips==0.1.4
markdown==3.5.1
markupsafe==2.1.3
matplotlib==3.7.4
mkl-fft==1.3.8
mkl-random==1.2.4
mkl-service==2.4.0
mpmath==1.3.0
networkx==3.1
numpy==1.22.0
nvidia-cublas-cu12==12.1.3.1
nvidia-cuda-cupti-cu12==12.1.105
nvidia-cuda-nvrtc-cu12==12.1.105
nvidia-cuda-runtime-cu12==12.1.105
nvidia-cudnn-cu12==8.9.2.26
nvidia-cufft-cu12==11.0.2.54
nvidia-curand-cu12==10.3.2.106
nvidia-cusolver-cu12==11.4.5.107
nvidia-cusparse-cu12==12.1.0.106
nvidia-nccl-cu12==2.18.1
nvidia-nvjitlink-cu12==12.3.101
nvidia-nvtx-cu12==12.1.105
oauthlib==3.2.2
omegaconf==2.3.0
opencv-python==4.5.4.58
packaging==21.3
pandas==2.0.3
parameterized==0.9.0
pillow==9.3.0
pims==0.6.1
pip==23.3.1
portalocker==2.8.2
protobuf==4.25.1
psutil==5.9.6
pyasn1-modules==0.3.0
pyasn1==0.5.0
pyparsing==3.1.1
python-dateutil==2.8.2
python-hostlist==1.21
pytorchvideo==0.1.5
pytz==2023.3.post1
pyyaml==6.0.1
regex==2023.10.3
requests-oauthlib==1.3.1
requests==2.31.0
rsa==4.9
sacremoses==0.1.1
scikit-learn==1.3.2
scipy==1.10.1
sentry-sdk==1.35.0
setproctitle==1.3.3
setuptools==68.0.0
six==1.16.0
slicerator==1.1.0
smmap==5.0.1
sympy==1.12
tabulate==0.9.0
tensorboard-data-server==0.7.2
tensorboard==2.14.0
termcolor==2.3.0
threadpoolctl==3.2.0
timm==0.4.12
tokenizers==0.10.3
torch==1.8.1
torchaudio==0.8.0a0+e4e171a
torchmetrics==1.2.1
torchvideo==0.0.0
torchvision==0.9.1
tqdm==4.50.2
transformers==4.10.2
triton==2.1.0
typing-extensions==4.7.1
tzdata==2023.3
urllib3==2.1.0
wandb==0.16.1
werkzeug==3.0.1
wheel==0.41.2
yacs==0.1.8
zipp==3.17.0

**Comment by ArtsiomWB** [comment]
Hi @mdivyanshu97, thank you for writing in, are you using any external library for your experiments such as pytorch or tensorflow?

**Comment by mdivyanshu97** [comment]
Thank you for the quick response. Yes I am using pytorch for my experiments.

**Comment by ArtsiomWB** [comment]
Thank you for your quick response, could you also please let me know what operating system you are using?

Trying to reproduce this behavior on my side. 

In the meantime, could you see if setting `WANDB_SILENT=True`, fixes your issue? We have docs on our env vars [here](https://docs.wandb.ai/guides/track/environment-variables#optional-environment-variables)

**Comment by ArtsiomWB** [comment]
Hi there, I wanted to follow up on this request. Please let us know if we can be of further assistance or if your issue has been resolved.

**Comment by ArtsiomWB** [comment]
Hi, since we have not heard back from you we are going to close this request. If you would like to re-open the conversation, please let us know!

**Comment by dmitryduev** [comment]
hey @mdivyanshu97, do you have `rocm-smi` installed on your cluster by any chance? we use that to monitor AMD GPU metrics, but since you have NVIDIA GPUs, as a quick workaround could you try uninstalling that tool?

**Comment by mdivyanshu97** [comment]
Hey, 

> Thank you for your quick response, could you also please let me know what operating system you are using?
> 
> Trying to reproduce this behavior on my side.
> 
> In the meantime, could you see if setting `WANDB_SILENT=True`, fixes your issue? We have docs on our env vars [here](https://docs.wandb.ai/guides/track/environment-variables#optional-environment-variables)

Hey, sorry for the late reply. I tried WANDB_SILENT but it doesn't fix the issue.
The issue only occurs on the cluster which uses the following OS.
 Operating System: CentOS Linux 7 (Core)

**Comment by mdivyanshu97** [comment]
> hey @mdivyanshu97, do you have `rocm-smi` installed on your cluster by any chance? we use that to monitor AMD GPU metrics, but since you have NVIDIA GPUs, as a quick workaround could you try uninstalling that tool?

Hey thank you for help. I will ask the IT department if they have any idea as I do not have sudo privileges on the cluster to uninstall this.

**Comment by gil2rok** [comment]
I also have this issue.

**Comment by ArtsiomWB** [comment]
@gil2rok do you have rocm-smi installed on your machine?

**Comment by gil2rok** [comment]
@ArtsiomWB sorry for the late reply. Yes I do have `rocm-smi` installed on my machine.

**Comment by dmitryduev** [comment]
hey @gil2rok, https://github.com/wandb/wandb/pull/6938 hopefully fixes it, but would appreciate it if you could verify that it actually does -- could you please install wandb from main (`pip install git+http://github.com/wandb/wandb@main)?
separately, out of curiosity, what do `rocm-smi -a --json` and `cat /sys/module/amdgpu/initstate` print for you on that system?

**Comment by gil2rok** [comment]
This new update fixed the bug for me. Feel free to close the pull request.

**Comment by gil2rok** [comment]
Actually, the error is still occuring. Instead of `WARNING: No AMD GPUs specified` being printed hundreds of times, it is now only printed once at the beginning of a wandb run.

**Comment by sydholl** [comment]
WandB Internal User commented: 
gil2rok commented: 
@ArtsiomWB sorry for the late reply. Yes I do have `rocm-smi` installed on my machine.

**Comment by collinmccarthy** [comment]
As of today (commit `c0a6547`) this issue is not fixed on our server, with "No AMD GPUs specified" being printed out every few seconds during training.

@dmitryduev Here's the information you requested.

```console
$ rocm-smi -a --json
WARNING: No AMD GPUs specified
{"system": {"Driver version": "6.3.6"}}

$ cat /sys/module/amdgpu/initstate
live
```

The server is managed by Slurm, and has a mix of AMD and NVIDIA GPUs, but in this allocation I've only requested an NVIDIA GPU.
